subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -537,6 +543,24 @@ private void writeAttributeBlock(
             List<Attribute<?>> providedAttributes = new ArrayList<>();
             Map<Attribute<?>, AttributeMatchDetails> bothAttributes = new LinkedHashMap<>();
             List<Attribute<?>> requestedAttributes = new ArrayList<>();
+
+            private final Comparator<Attribute<?>> byAttributeName = Comparator.comparing(Attribute::getName);
+
+            private final Comparator<? super Map.Entry<Attribute<?>, AttributeMatchDetails>> byEntryAttributeName = Comparator.comparing(e -> e.getKey().getName());
+
+            private final Supplier<TreeSet<Map.Entry<Attribute<?>, AttributeMatchDetails>>> sortedEntrySet = () -> new TreeSet<>(byEntryAttributeName);
+
+            List<Attribute<?>> getProvidedAttributesSorted() {
+                return providedAttributes.stream().sorted(byAttributeName).collect(Collectors.toList());
+            }
+
+            Set<Map.Entry<Attribute<?>, AttributeMatchDetails>> getBothAttributesEntrySetSorted() {
+                return bothAttributes.entrySet().stream().collect(Collectors.toCollection(sortedEntrySet));
+            }
+
+            List<Attribute<?>> getRequestedAttributesSorted() {
+                return requestedAttributes.stream().sorted(byAttributeName).collect(Collectors.toList());
+            }
```

Comment:
It's probably better to collect directly to a `TreeSet`/`TreeMap` rather than re-sort these a second time.

Reply:
Can do - I wasn't sure if preserving their original ordering in `AttributeBuckets` had any value.